### ISSUE TITLE
`user` setting causes master to die

### ISSUE THREAD

**Issue by kreichgauer** [LearnBoost/cluster#61]
When i put `.set('user', 'foo')` in my application, and start my cluster with `nohup ./server.js &`, the master will die after spawning the worker processes. I haven't really figured out why, but observed it with node 0.4.2 under Linux. Launching `server.js` as `foo` and removing the user config directive fixes the problem.

**Comment by tj** [comment]
Can you add the "debug()" plugin and paste the output?

**Comment by kreichgauer** [comment]
Looks like it's keeping the wrong master PID...

```
$ sudo /usr/local/bin/node server.js > node.out 2>&1 &
[8] 10095
$ ./server.js status

  worker 0 10099 alive
  worker 1 10101 alive
  master 10038 dead

  debug - exit
$ ps aux|grep node
(PIDs are 10095 for the master, 10099 and 10101 for the slaves)
$ cat node.out 
  info - master started
  info - worker 0 spawned
  info - worker 1 spawned
$ ./server.js stop

node.js:116
        throw e; // process.nextTick error, or 'error' event on first tick
        ^
Error: ESRCH, No such process
```

**Comment by tj** [comment]
hmm I tried similar and cant seem to reproduce it

**Comment by kreichgauer** [comment]
Did you make sure to `.set('user', <user>) to a different user than the one that spawns the master process?

**Comment by tj** [comment]
yeah it's all good here

**Comment by kreichgauer** [comment]
Strange, I can reproduce it on 2 different machines, both Linux. Which node version did you try?

**Comment by tj** [comment]
0.4.2, I'm on osx though

**Comment by tj** [comment]
any luck?

**Comment by kreichgauer** [comment]
Just realized I'm still stuck on cluster 0.4.0. Might that be the problem? Will try 0.4.2 soon.

**Comment by tj** [comment]
ah, yeah that would be it, `user` was fixed in 0.4.2
